manifests/grafana.pp

```diff
@@ -21,6 +21,16 @@
       require => Package['grafana'],
       before  => Service['grafana-server'],
     }
+  }
+
+  if $puppet_metrics_dashboard::enable_ldap_auth {
```

Comment:
We can't add an `if` here, because the else below it is already part of the first if statement above.  I'd suggest approaching this by setting a new variable `$ldap_config` with the keys / values you need and then doing an additional merge below (see line 48) so that everything ends up in the `$_grafana_cfg` var.

You'll also need to update spec tests in `spec/classes/grafana_spec.rb` because we don't currently expect ldap config to be there.

Reply:
Just pushed some updates to address this.